src/Integrator/BDHI/DoublyPeriodic/DPStokesSlab.cuh

```diff
@@ -96,6 +96,44 @@ namespace uammd{
 	return {particleVelocities, particleAngularVelocities};
       }
 
+      // compute average velocity in the x direction as a function of z
+      template<class PosIterator, class ForceIterator>
+      std::vector<double>
+      computeAverageVelocity(PosIterator pos, ForceIterator forces,
+			     int numberParticles, cudaStream_t st = 0){
```

Comment:
If you want to make this function valid for X or Y add an optional parameter here, ```, int direction = 0, cudaStre... ```

Reply:
Done.